internal/driver/ldap/main.go

```diff
@@ -0,0 +1,112 @@
+package ldap
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"regexp"
+	"text/template"
+
+	client "github.com/go-ldap/ldap/v3"
+	"github.com/peeley/carpal/internal/config"
+	"github.com/peeley/carpal/internal/driver"
+	"github.com/peeley/carpal/internal/resource"
+	"gopkg.in/yaml.v3"
+)
+
+type LdapClient interface {
+	Bind(string, string) error
+	Close() error
+	Search(*client.SearchRequest) (*client.SearchResult, error)
+}
+
+type ldapDriver struct {
+	Configuration config.Configuration
+	Template      *template.Template
+	ClientFunc    func() (LdapClient, error)
+}
+
+func NewLDAPDriver(conf config.Configuration) driver.Driver {
+	d := ldapDriver{
+		Configuration: conf,
+	}
+	d.Template = template.Must(template.ParseFiles(conf.LDAPConfiguration.Template))
+	d.ClientFunc = func() (LdapClient, error) {
+		return client.DialURL(conf.LDAPConfiguration.URL)
+	}
+	return d
+}
+
+func (d ldapDriver) GetResource(name string) (*resource.Resource, error) {
+	var resource resource.Resource
+
+	re, err := regexp.Compile("(.+):([^@]+)")
```

Comment:
Below, we're explicitly checking that this first capture group is `"acct"`. Maybe just check for that in the regex?
```suggestion
	re, err := regexp.Compile("acct:([^@]+)")
```

Reply:
I split it up to allow handling different resource types later but I'm fine with hard-coding acct for now and dealing with other resources later/if ever.

Reply:
Swapped the regex.